server/ladder_service.py

```diff
@@ -500,6 +500,11 @@ def make_game_options(player: Player) -> GameLaunchOptions:
                     "Ladder game failed to start! %s setup timed out",
                     game
                 )
+            elif isinstance(e, asyncio.CancelledError):
```

Comment:
Hmm, I think using set_exception with a custom error would be better because cancelled error could be raised for many reasons.